### DISCUSSION TITLE
Unable to restore Sql Server database in codespace

### DISCUSSION THREAD

**Discussion by pedro-gilmora** [community/community#38630] [Codespaces]
I have the following script with its corresponding backup file

**Restore.SQL**

```sql
USE [master];

RESTORE DATABASE [Imports]
FROM DISK = '/workspaces/Imports/.devcontainer/mssql/Imports.bak'
WITH MOVE 'Imports' TO '/var/opt/mssql/data/Imports.mdf',
MOVE 'Imports_log' TO '/var/opt/mssql/data/Imports_log.ldf',
FILE = 1, STATS = 10
GO
```
In command line I execute this line

`sqlcmd -S localhost -U sa -P MSSqlPasswd -d master -i /workspaces/Imports/.devcontainer/mssql/Restore.SQL`

but I get this error:

>Msg 3201, Level 16, State 2, Line 3
Cannot open backup device '/workspaces/Imports/.devcontainer/mssql/Imports.bak'. Operating system error 2.
(The system cannot find the file specified.).

>Msg 3013, Level 16, State 1, Line 3
`RESTORE DATABASE` is terminating abnormally

As you can see the SQL script and backup file are in the same folder (`.devcontainer/mssql`) with `chmod 777` permissions

A small research thrown a possibility that `mssql` user might not have right to find the backup file due to permissions. Not even have access to `/var/opt/mssql/data or backup`, like they dont't exist

Also I moved it to a location that might be more "visible" or globally available like `/home/shared` and all permissions were granted

**Comment by ossanna16** [comment]
Hi there @pedro-gilmora and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
